docs/advanced/porting_guide.md

```diff
@@ -36,6 +36,52 @@ Add the target description to ```mbed-os\targets\targets.json```:
 
 See the [mbed Target Documentation](mbed_targets.md) for more details on what this definition means.
 
+### Coding requirements
+
+All code submitted as part of a port must compile with GCC ARM Embedded, ARM Compiler 5 and IAR EWARM. Flags that port code must compile with for each compiler are listed below. The profiles used to build mbed OS are subsets of each of these flags with exactly one optimization setting.
+
+#### GCC ARM Embedded
+
+Code in a port must be compatible with the `softfp` float ABI and compile without warnings or errors using GCC ARM Embedded with the following flags.
```

Comment:
That implies that ports are allowed to not use GCC ARM Embedded. That is not true.